io/net/bridge.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: Harish Sriram <harish@linux.vnet.ibm.com>
+# Bridge interface test
+
+
+import netifaces
+from avocado import main
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+
+
+class Bridging(Test):
+    '''
+    Test bridge interface
+    '''
+
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        smm = SoftwareManager()
+        if not smm.check_installed("bridge-utils") and not smm.install("bridge-utils"):
+            self.cancel("bridge-utils package is need to test")
+
+        self.host_interface = self.params.get("host_interface",
+                                              default=None)
+        if not self.host_interface:
+            self.cancel("User should specify host interface")
+
+        interfaces = netifaces.interfaces()
+        if self.host_interface not in interfaces:
+            self.cancel("Interface is not available")
+
+        self.peer_ip = self.params.get("peer_ip", default=None)
+        if not self.peer_ip:
+            self.cancel("User should specify peer IP")
+
+        cmd = "ip addr show %s | sed -nr 's/.*inet ([^ ]+)."\
+            "*/\\1/p'" % self.host_interface
+        self.cidr = process.system_output(
+            '%s' % cmd, shell=True)
+        cmd = "route -n | grep %s | grep -w UG | awk "\
+            "'{ print $2 }'" % self.host_interface
+        self.gateway = process.system_output(
+            '%s' % cmd, shell=True)
+        cmd = "ip addr show %s | grep inet | grep brd | "\
+            "awk '{ print $4 }'" % self.host_interface
+        self.broadcast = process.system_output(
+            '%s' % cmd, shell=True)
+
+    def test(self):
+        '''
+        Bridge Test
+        Set up the ethernet bridge configuration in the linux kernel
+        '''
+        process.system('brctl addbr br0', sudo=True)
```

Comment:
raising CmdError i think not good i think we have proper validation for every command if command output is failed (not matched with desired  output ) need to fail the test